ph_logger.c

```diff
@@ -222,6 +229,74 @@ static char *strnchr(char *src, char ch, int len)
 	return NULL;
 }
 
+struct ph_logger_file *_search_log_file(const char *path)
+{
+	struct ph_logger_file *f, *tmp;
+	dl_list_for_each_safe(f, tmp, &ph_logger.files, struct ph_logger_file, list)
+	{
+		if (pv_str_matches(path, strlen(path), f->path, strlen(f->path)))
+			return f;
+	}
+
+	return NULL;
+}
+
+static void _save_log_file_pos(off_t pos, const char *path)
+{
+	// try to get file info from memory
+	struct ph_logger_file *f;
+	f = _search_log_file(path);
+	// if not, we create a new entry
+	if (!f) {
+		pv_log(DEBUG, "log file '%s' not yet stored in memory. Saving new file...", path);
+		f = calloc(1, sizeof(struct ph_logger_file));
+		f->path = strdup(path);
+		dl_list_add(&ph_logger.files, &f->list);
+	}
+	if (!f)
+		pv_log(ERROR, "could not initialize log file: %s", strerror(errno));
+
+	// update pos value in memory
+	f->pos = pos;
```

Comment:
This assignment could be an error if both `if (!f)` are true. So I guess we need to include a `return` in the second `if` after the of the `pv_log` call